### PR TITLE
[PRPP-999] Browser back button on PDF full screen does not replicate breadcrumb

### PR DESCRIPTION
**Pull request by abid-nhs**
**Overview**
When using the back within the browser the buttons are working in a way that is not user friendly following the breadcrumb pattern.

**Current Behaviour**
When pressing back button on full screen reverts to the patient confirmation screen
This is due to how the components work however isn’t user friendly

**Expected Behaviour**
Back button should take to the frame before

### CODE REVIEW

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx [hunk 1]
```diff
@@ -60,6 +60,26 @@ function LloydGeorgeViewRecordStage({
...
+            }
+        };
+
+        window.addEventListener('popstate', handleBackButton);
+
+        return () => {
+            window.removeEventListener('popstate', handleBackButton);
+        };
+    }, [fullScreen]);
+
```
[the remark this answers is not in the document]
Comment (abid-nhs) [reply app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx:80]: This has been added as per your request. Thanks

**Review by RioKnightleyNHS — APPROVED** [review]

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx [hunk 2]
```diff
@@ -143,6 +143,48 @@ describe('LloydGeorgeViewRecordStage', () => {
...
+        act(() => {
+            window.dispatchEvent(new PopStateEvent('popstate'));
+        });
+
+        await waitFor(() => {
+            expect(screen.getByText('Lloyd George record')).toBeInTheDocument();
+        });
+    });
+
+    it('updates history correctly when exiting full screen using the UI back button', async () => {
```
Comment (MohammadIqbalAD-NHS) [inline app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx:166]: The test on line 124 mentions "Exit full screen link" whereas line 166 mentions "UI back button". As far as I'm aware, they both are referring to the same button. Could we name these identically for consistency?

Comment (abid-nhs) [reply app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx:166]: ah yes. Good catch. Thanks for that. will make them consistent.

**Review by MohammadIqbalAD-NHS — APPROVED** [review]

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx [hunk 3]
```diff
@@ -61,20 +61,49 @@ function LloydGeorgeViewRecordStage({
...
+                window.history.back();
             }
         };
 
-        window.addEventListener('popstate', handleBackButton);
+        const handleBrowserForwardButton = () => {
+            if (!fullScreen) {
+                setFullScreen(true);
+            }
+        };
```
Comment (MohammadIqbalAD-NHS) [inline app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx:103]: Considering these functions are purely doing IF statements on fullscreen, could they not be combined into 1 function?

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx [hunk 4]
```diff
@@ -183,6 +183,98 @@ describe('LloydGeorgeViewRecordStage', () => {
...
+        expect(backSpy).toHaveBeenCalled();
+
+        act(() => {
+            window.history.forward();
+        });
+
+        await waitFor(() => {
+            expect(screen.getByText('Exit full screen')).toBeInTheDocument();
+        });
+        // expect(forwardSpy).toHaveBeenCalled();
```
Comment (MohammadIqbalAD-NHS) [inline app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx:220]: Commented out code needs removing

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx (hunk 3 above)
Comment (abid-nhs) [reply app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx:103]: They can, but they have been seperated like this to make it clear which one is to be done in back button click and which logic is to be done for forward button click

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx (hunk 4 above)
Comment (abid-nhs) [reply app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.test.tsx:220]: ah, I'll remove this now. Thanks

File: app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx (hunk 3 above)
Comment (MohammadIqbalAD-NHS) [reply app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx]: Makes sense. I guess it's a matter of preference, whether having 2 functions justify the better readability or not. But I'm fine with that reasoning.

**Comment by MohammadIqbalAD-NHS** [comment]
There are Cypress Tests which are failing.

**Pull request closed without merging**